### DISCUSSION TITLE
Additional automergeType "pr-title-tag"

### DISCUSSION THREAD

**Discussion by Pfoerd** [renovatebot/renovate#29247] [Suggest an Idea]
Some "merge bots" like the Auto-Merge plugin for Bitbucket Datacenter (https://docs.mibexsoftware.com/codeowners/auto-merge) are using a tag in the pr title to control auto-merging. E.g. the Auto-Merge plugin prefixes all pr titles that should be automatically merged with [AUTO] , e.g. "[AUTO] build(*): bump libA to v2.0.0"

So what I'm requesting is to add a new automergeType "pr-title-tag" and a new configuration property "automergeTitleTag" to enable such merge bots.

Of course, Renovate's matching algorithms for branches/PRs shouln't be affected by these tags.

**Comment by rarkins** [comment]
Do you need this title changed only after branch tests have passed, or set at PR creation time?

**Reply by Pfoerd** [reply]
In case of the Auto-Merge plugin the [AUTO] tag can be added at any time, thus at creation time and also any time after creation.

**Reply by rarkins** [reply]
I was meaning to ask when do _you_ need Renovate to add that title change. e.g. do you want Renovate to create a PR and add the tag immediately (before tests are completed) or wait until a future run once tests have passed

**Reply by Pfoerd** [reply]
I'm asking against the background that we want to use the aforementioned Auto-Merge plugin, so both would be okay for me (as the Auto-Merge bot only merges PRs that passed all checks in Bitbucket)

**Reply by rarkins** [reply]
Can you try configuring `"commitMessageSuffix": "[AUTO]"` and seeing if that satisfies your requirement for automerging? It should append the text to commit messages as well as PR titles.

**Reply by Pfoerd** [reply]
I had already considered this approach but the problem here is that having the [AUTO] tag in the commit message is often at least not very elegant and in my specific case also not possible because there are special commit message constraints enforcing no special characters like "[" "]" in the commit message.

**Reply by rarkins** [reply]
In this case we should look into using prTitle alone. If you are able to _test_ conveniently, could you try if `"prTitle": "{{ commitMessage }} [AUTO]"` works?

**Reply by Pfoerd** [reply]
Changing the prTitle was my first idea but I had doubts because of various hints in the documentation that this way is deprecated and that the prTitle should not actually be changed. But okay than I give it a try 👍 . 
So you don't expect any of the problems mentioned in the documentation regarding the matching algorithms when changing the PR title the way you suggest?

**Reply by rarkins** [reply]
The theoretical problem is that if you had identical upgrades but one with automerge and one without then they'd be treated as separate PRs.

e.g. let's say you had some old PR without the [AUTO] which you closed/ignored. You'll get a new PR now in such a case because the PR title has changed

**Reply by Pfoerd** [reply]
Okay so that means the main issue is that closed PRs could be raised again? I think this is not too serious a problem (it would be more problematic if updates are "missed")

**Reply by Pfoerd** [reply]
So I tested it. Changing the PR title via `prTitle` property seems to work in general. Strange thing is that the [AUTO] tag was lowercased, so the resulting PR-Title for `"prTitle": "[AUTO] {{ commitMessage }}"` now are PRs with titles like "[auto] build(deps): ...". This prevents auto-merging as the Auto-Merge Plugin is case-sensitive.

Do you know why the tag-text is lowercased and how this can be prevented?

**Reply by rarkins** [reply]
This would likely be due to semantic commit logic, where the part before : is meant to be lower cased

**Reply by Pfoerd** [reply]
so do you think setting [`commitMessageLowerCase`](https://docs.renovatebot.com/configuration-options/#commitmessagelowercase) to `never` could be a reasonable fix for this?

**Reply by rarkins** [reply]
I'm not sure if that works in the left hand side of the : too

**Reply by Pfoerd** [reply]
I have just tried it and setting `"commitMessageLowerCase": "never"` did the job. So it seems customizing the prTitle seems to be a reasonable workaround for this case.

Do you think it could make sense anyway to add an additional option to renovates automerge function to support this more "native" and treat it as a first-class citizen?

**Reply by rarkins** [reply]
Only if there was wide demand for this, or by a Mend customer. Otherwise we prefer to avoid adding options/complexity if a feature is only needed by a very small subset of users. We can leave this Discussion open to see if it's a popular use case over time
